jdaviz/app.py

```diff
@@ -721,52 +721,28 @@ def get_subsets_from_viewer(self, viewer_reference, data_label=None, subset_type
                                          cls=None)
         regions = {}
 
-        def _get_all_subregions(mask, spec_axis_data):
-            """
-            Return all subregions within a subset.
-
-            Parameters
-            ----------
-            mask : list
-                List of indices in spec_axis_data that are part of the subset.
-            spec_axis_data : list
-                List of spectral axis values.
-            Returns
-            -------
-            combined_spec_region : `~specutils.SpectralRegion`
-                SpectralRegion object containing all subregions of the subset.
-            """
-            if len(mask) == 0:
-                # Mask should only be 0 if ApplyROI is used to incorrectly
-                # create subsets via the API
-                raise ValueError("Mask has length 0, ApplyROI may have been used incorrectly")
-
-            current_edge = 0
+        def _get_all_subregions(spectral_axis_units, subset_label):
             combined_spec_region = None
-            for index in range(1, len(mask)):
-                # Find spot where mask == True is for a different region of the subset
-                # i.e. mask = [0, 1, 4, 5]
-                # mask[2] != mask[1] + 1
-                if mask[index] != mask[index - 1] + 1:
-                    subset_region = spec_axis_data[mask[current_edge]: mask[index - 1] + 1]
-                    if not combined_spec_region:
-                        combined_spec_region = SpectralRegion(min(subset_region),
-                                                              max(subset_region))
-                    else:
-                        combined_spec_region += SpectralRegion(min(subset_region),
-                                                               max(subset_region))
-                    current_edge = index
-
-            # Get last region within the subset
-            if current_edge != index:
-                subset_region = spec_axis_data[mask[current_edge]: mask[index] + 1]
-                # No if check here because len(mask) must be greater than 1
-                # so combined_spec_region will have been instantiated in the for loop
-                if combined_spec_region is None:
-                    combined_spec_region = SpectralRegion(min(subset_region), max(subset_region))
-                else:
-                    combined_spec_region += SpectralRegion(min(subset_region), max(subset_region))
+            try:
+                subset_plugin = self._jdaviz_helper.plugins['Subset Tools']
+            except ValueError as e:
+                print(e)
+                return None
```

Comment:
Can we put in a hack to do something like this?

* If it is only of the configs we support, then when it is not found, it would be found later, so nothing to do here.
* If it is not one of the configs we support, what do we care? Just make a new instance and hope for the best.

Reply:
Unless you can find out in the config yaml that is cached somewhere what plugin is supposed to load?

Reply:
Ah, yeah, you can dig at `app._loaded_configuration` if you really need to know in advance.

Reply:
But then if one day, we also let people load in plugin after-the-fact from GUI, then 🔥

Reply:
My suggestion (if we're not refactoring to put the code in functions outside the plugin) would be a cached property for the subset_plugin that does `return self._jdaviz_helper.plugins.get('Subset Tools', SubsetTools(app=self))`.  This would instantiate an instance of the plugin (but not show it in the tray) if it isn't already loaded, and would only do that the first time this is called.